File: Labs/FinalLab/native/VolunteersApp/app/src/main/java/com/example/ma/volunteersapp/EditActivity.java
package com.example.ma.volunteersapp;

import android.content.ContentValues;
import android.content.Intent;
import android.database.Cursor;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.Toast;

public class EditActivity extends AppCompatActivity {

    DBHelper helper;
    EditText TxName, TxPhone;
    long id;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit);

        helper = new DBHelper(this);

        id = getIntent().getLongExtra(DBHelper.row_id, 0);

        TxName = (EditText)findViewById(R.id.txName_Edit);
        TxPhone = (EditText)findViewById(R.id.txPhone_Edit);

        getData();
    }

    private void getData() {
        System.out.println(id);
        Cursor cursor = helper.oneData(id);
        System.out.println(cursor);
        if(cursor.moveToFirst()){
            String name = cursor.getString(cursor.getColumnIndex(DBHelper.row_name));
            String phone = cursor.getString(cursor.getColumnIndex(DBHelper.row_phone));

            TxName.setText(name);
            TxPhone.setText(phone);
        }
    }

    public boolean onCreateOptionsMenu (Menu menu){
        getMenuInflater().inflate(R.menu.edit_menu, menu);
        return true;
    }

    public boolean onOptionsItemSelected(MenuItem item){
        switch (item.getItemId()){
            case R.id.save_edit:
                String name = TxName.getText().toString().trim();
                String phone = TxPhone.getText().toString().trim();

                ContentValues values = new ContentValues();
                values.put(DBHelper.row_name, name);
                values.put(DBHelper.row_phone, phone);

                if (name.equals("") && phone.equals("")){
                    Toast.makeText(EditActivity.this, "Nothing to save", Toast.LENGTH_SHORT).show();
                }else {
                    helper.updateData(values, id);
                    int idd = (int)id;
                    Intent newIntent = new Intent();
                    newIntent.putExtra("ID", idd);
                    newIntent.putExtra("NAME", name);
                    newIntent.putExtra("PHONE", phone);
                    setResult(2, newIntent);
                    Toast.makeText(EditActivity.this, "Saved", Toast.LENGTH_SHORT).show();
                    finish();
                }
        }
        switch (item.getItemId()){
            case R.id.delete_edit:
                Cursor cur = helper.oneData(id);
                cur.moveToFirst();
                int idd = (int)id;
                Intent newIntent = new Intent();
                newIntent.putExtra("ID", idd);
                setResult(3, newIntent);
                Intent idvolunteers = new Intent(EditActivity.this, DeleteActivity.class);
                idvolunteers.putExtra(DBHelper.row_id, id);
                startActivityForResult(idvolunteers,3);
                finish();

        }
        return super.onOptionsItemSelected(item);
    }
}

File: Labs/Lab1/ManageVolunteersNativeApplicationKotlin/app/src/main/java/com/example/myapplication/EditActivity.kt
package com.example.myapplication

import android.content.ContentValues
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class EditActivity : AppCompatActivity() {
    var helper: DBHelper? = null
    var TxTitle: EditText? = null
    var TxDetail: EditText? = null
    var id: Long = 0
    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit)
        helper = DBHelper(this)
        id = getIntent().getLongExtra(DBHelper.row_id, 0)
        TxTitle = findViewById(R.id.txTitle_Edit) as EditText?
        TxDetail = findViewById(R.id.txDetail_Edit) as EditText?
        data
    }

    private val data: Unit
        private get() {
            val cursor = helper!!.oneData(id)
            if (cursor.moveToFirst()) {
                val title = cursor.getString(cursor.getColumnIndex(DBHelper.row_title))
                val detail = cursor.getString(cursor.getColumnIndex(DBHelper.row_note))
                TxTitle!!.setText(title)
                TxDetail!!.setText(detail)
            }
        }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        getMenuInflater().inflate(R.menu.edit_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.save_edit -> {
                val title = TxTitle!!.text.toString().trim { it <= ' ' }
                val detail = TxDetail!!.text.toString().trim { it <= ' ' }
                val values = ContentValues()
                values.put(DBHelper.row_title, title)
                values.put(DBHelper.row_note, detail)
                if (title == "" && detail == "") {
                    Toast.makeText(this@EditActivity, "Nothing to save", Toast.LENGTH_SHORT).show()
                } else {
                    helper!!.updateData(values, id)
                    Toast.makeText(this@EditActivity, "Saved", Toast.LENGTH_SHORT).show()
                    finish()
                }
            }
        }
        when (item.itemId) {
            R.id.delete_edit -> {
                val cur = helper!!.oneData(id)
                cur.moveToFirst()
                val idnotes = Intent(this@EditActivity, DeleteActivity::class.java)
                idnotes.putExtra(DBHelper.row_id, id)
                startActivity(idnotes)
                finish()
            }
        }
        return super.onOptionsItemSelected(item)
    }
}